jupyter_server/extension/application.py

```diff
@@ -175,13 +175,20 @@ def _validate_extension_name(self, value):
             return value
         raise ValueError("Extension name must be a string, found {type}.".format(type=type(value)))
 
+    @classmethod
+    def _jupyter_server_extension_paths(cls):
+        return [{
+            "module": cls.__module__.split('.')[0],
+            "app": cls
+        }]
+
```

Comment:
New method that builds extension metadata.

Reply:
I have a few thoughts on this, but I think it's worth discussing in tomorrow's meeting. After that discussion, I'll follow up here with notes.

Reply:
I think it makes the most sense to explicitely have extensions authors write this function, so I've removed this from ExtensionApps.